schemas/iso19110/src/main/plugin/iso19110/process/add-columns-from-csv.xsl

```diff
@@ -0,0 +1,157 @@
+  ~ General Public License for more details.
+  ~
+  ~ You should have received a copy of the GNU General Public License
+  ~ along with this program; if not, write to the Free Software
+  ~ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
+  ~
+  ~ Contact: Jeroen Ticheler - FAO - Viale delle Terme di Caracalla 2,
+  ~ Rome - Italy. email: geonetwork@osgeo.org
+  -->
+
+<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
+                xmlns:gfc="http://www.isotc211.org/2005/gfc"
+                xmlns:geonet="http://www.fao.org/geonetwork"
+                xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                xmlns:xs="http://www.w3.org/2001/XMLSchema"
+                xmlns:gco="http://www.isotc211.org/2005/gco"
+                xmlns:srv="http://www.isotc211.org/2005/srv"
+                xmlns:gmd="http://www.isotc211.org/2005/gmd"
+                version="2.0"
+                exclude-result-prefixes="#all">
+
+  <xsl:import href="../../iso19139/process/process-utility.xsl"/>
+
+  <!-- i18n information -->
+  <xsl:variable name="csv-info-loc">
+    <msg id="a" xml:lang="eng">Add columns from CSV</msg>
+    <msg id="a" xml:lang="fre">Ajouter les colonnes à partir d'un fichier CSV</msg>
+    <msg id="a" xml:lang="dut">Add columns from CSV</msg>
+  </xsl:variable>
+
+  <!-- Process parameters and variables-->
+  <xsl:param name="replaceColumns" select="'1'"/>
+  <xsl:param name="columnListAsCsv" select="''"/>
+  <xsl:param name="columnListSeparator" select="';'"/>
+
+
+  <xsl:template name="list-add-columns-from-csv">
+    <suggestion process="add-columns-from-csv"/>
+  </xsl:template>
+
+
+  <!-- Analyze the metadata record and return available suggestion
+    for that process -->
+  <xsl:template name="analyze-add-columns-from-csv">
+    <xsl:param name="root"/>
+
+    <suggestion process="add-columns-from-csv" id="{generate-id()}"
+                category="fcat" target="gfc:FC_FeatureType">
+      <name>
+        <xsl:value-of select="geonet:i18n($csv-info-loc, 'a', $guiLang)"/>
+      </name>
+      <operational>true</operational>
+      <params>{
+        "replaceColumns":{"type":"boolean", "defaultValue":"1"},
+        "columnListSeparator":{"type":"string", "defaultValue":"<xsl:value-of select="$columnListSeparator"/>"},
+        "columnListAsCsv":{"type":"textarea", "defaultValue":"SHORT_NAME;DESCRIPTION;TYPE;CARDINALITY_MIN;;CARDINALITY_MAX"}
+      }</params>
+    </suggestion>
+  </xsl:template>
+
+  <xsl:template match="gfc:FC_FeatureType" priority="99">
+
+    <xsl:copy>
+      <xsl:apply-templates select="@*"/>
+      <xsl:apply-templates select="*[name() != 'gfc:carrierOfCharacteristics']"/>
+
+      <xsl:if test="$replaceColumns != '1'">
+        <xsl:apply-templates select="gfc:carrierOfCharacteristics"/>
+      </xsl:if>
+
+      <xsl:variable name="lines"
+                    select="tokenize($columnListAsCsv, '\n')"/>
+
+      <xsl:for-each select="$lines">
```

Comment:
Currently empty lines are processed, adding fields with empty values. Similar issue in the suggestions for field codelists.